profiling/src/lib.rs

```diff
@@ -290,6 +290,17 @@ extern "C" fn minit(_type: c_int, module_number: c_int) -> ZendResult {
     #[cfg(feature = "exception_profiling")]
     exception::exception_profiling_minit();
 
+    // There are a few things which need to do something on the first rinit of
+    // each minit/mshutdown cycle. In Apache, when doing `apachectl graceful`,
+    // there can be more than one of these cycles per process.
+    // Re-initializing these on each minit allows us to do it once per cycle.
+    // This is unsafe generally, but all SAPIs are supposed to only have one
+    // thread alive during minit, so it should be safe here specifically.
+    unsafe {
+        ZAI_CONFIG_ONCE = Once::new();
+        RINIT_ONCE = Once::new();
+    }
+
```

Comment:
I would think that resetting the `Once` should go to `mshutdown`. It should already be initialised in the `static` below, so this would be a second initialisation. Additionally we could check for the Apache SAPI and only do it if we detect Apache, but maybe that is not worth it, as a `Once::new()` seems cheap and other SAPI's (now or in the future) might do the same ;-)

Reply:
Technically, you are right. I'm not sure why, but it makes me more comfortable to put it in startup than shutdown. I'll think about it more and see if I can find a real justification or not.

Reply:
"more comfortable" as in better save than sorry?
I can totally see that, let's leave it as is.